sdk/iot/core/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # SPDX-License-Identifier: MIT
 
-cmake_minimum_required (VERSION 3.12)
+cmake_minimum_required (VERSION 3.10)
```

Comment:
Is it required to state the min version in each CMakeLists.txt file or is having one in the root enough? If it's needed, is there a way to capture the version in some constant and reuse that in all the CMakeLists, so we can avoid having to update it in all the places?

Also, what factors go into deciding the cmake version required? Why 3.10 and not something older/newer? Why was 3.12 too high?

Reply:
The latest version of cmake via apt-get on Ubuntu 18.04 is 3.10.2. So developing on that platform can be a pain. As for the cmake_version it seems it might only be needed on the top level.
[Link here](https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_VERSION.html)